docs/howto_contribute.md

```diff
@@ -0,0 +1,396 @@
+
+#### Example Playbook
+
+In the example playbook, we are going to copy a couple of Dockerfiles to the
+host and then run the `docker build` command using those files.  Afterwards,
+we will remove the Docker images that have been built.
+
+Since we are going to be copying multiple Dockerfiles, we will need to create
+a `files` directory to hold them.  See below for the directory structure and
+some simple example Dockerfiles.
+
+```bash
+$ mkdir -p tests/docker-build/files/
+$ cat tests/docker-build/files/centos-httpd-Dockerfile
+FROM registry.centos.org/centos/centos:7
+RUN yum -y install httpd
+
+$ cat tests/docker-build/files/fedora-httpd-Dockerfile
+FROM registry.fedoraproject.org/fedora:25
+RUN dnf -y install httpd
+```
+
+With the files in place, we can edit our playbook at `tests/docker-build/main.yml`.
+While it is not required to use the name `main.yml`, it is the standard file name
+we have used thusfar.
+
+```yaml
+---
+- name: Docker Build
+  hosts: all
+  become: yes
+
+  tags:
+    - docker_build
+
+  pre_tasks:
+    - name: Make temp directory to hold Dockerfiles
+      command: mktemp -d
+      register: temp_dir
+
+    - name: Copy Dockerfiles to temp directory
+      synchronize:
+        src: files/
+        dest: "{{ temp_dir.stdout }}/"
+        recursive: yes
+
+  roles:
+    - role: ansible_version_check
+      avc_major: "2"
+      avc_minor: "2"
+      tags:
+        - ansible_version_check
+
+    - role: docker_build
+      build_path: "{{ temp_dir.stdout }}"
+      dockerfile_name: "centos-httpd-Dockerfile"
+      image_name: "centos-httpd"
+      tags:
+        - centos_docker_build
+
+    - role: docker_build
+      build_path: "{{ temp_dir.stdout }}"
+      dockerfile_name: "fedora-httpd-Dockerfile"
+      image_name: "fedora-httpd"
+      tags:
+        - fedora_docker_build
+
+  post_tasks:
+    - name: Remove all docker images
+      shell: docker rmi -f $(docker images -qa)
+```
+
+Here we've made a temporary directory, copied the Dockerfiles to the directory,
+built the images using the `docker_build` role from the earlier example, and
+then cleaned up after the test.
+
+Note, we used the `ansible_version_check` role to verify that the version of
+Ansible being used to execute the playbook is what we currently support.  This
+is suggested for all playbooks.
+
+Each role is tagged in the same way the whole playbook is tagged - to enable
+users to skip certain roles during test execution.
+
+Additionally, it is good practice to clean up any artifacts from the test
+which may interfere with other tests that are run afterwards.
+
+### Testing Your Playbook
+
+After you have finished with your playbook, it is important to test that it can
+successfully run against the Atomic Host variant of CentOS, Fedora, and RHEL.
+If one of those platforms does not run successfully, you'll need to make the
+necessary adjustments before submitting a pull request with your changes.
+
+### Documenting Your Playbook
```

Comment:
Ahh here it is.  Suggest moving this up, as the **first** item.  Emphasizing the big-picture planning and docs maintence, rather than leave it to an after thought.  Most people will read the example playbook section (above), and stop there if accidentally works :S